### PR TITLE
feat: add X button to clear search bar

### LINKED ISSUE #345: feature: Create a "cancel" button to clear `input` values
## Summary

We'll need a "clear" button added to some inputs, like the Search.

- [Figma Link](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/OpenSauced-%E2%80%94-Insights-(insights.opensauced.pizza)?node-id=3753%3A4358)

### Features
- When press <kbd>ESC</kbd> **while input is in focus:** clear the field
- Have an <kbd>X</kbd> or <kbd>clear</kbd> button at the end of the input, when clicked, it clears the field. 

Screen Shot 2022-09-21 at 5 21 02 PM

---

### Original Discussion:

Making a note here for us to create a "cancel" feature later on.

_Originally posted by @pixelsbyeryc in https://github.com/open-sauced/insights/pull/337#discussion_r967618851_

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

Adds an "X" button that clears the search bar.
## Related Tickets & Documents

closes #345
## Mobile & Desktop Screenshots/Recordings

image

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/atoms/Search/search.tsx [hunk 1]
```diff
@@ -15,6 +16,10 @@ const Search = ({ placeholder, name, value, autoFocus, className, onSearch }: Se
     onSearch?.(search);
   };
 
+  const emptySearch = () => {
+    setSearch("");
```
Comment (brandonroberts) [inline components/atoms/Search/search.tsx:20]: Call the provided `onSearch` function so it triggers an update for the parent component also

```suggestion
    setSearch("");
    onSearch?.("");
```

File: components/atoms/Search/search.tsx [hunk 2]
```diff
@@ -37,6 +42,8 @@ const Search = ({ placeholder, name, value, autoFocus, className, onSearch }: Se
           }
         }}
       />
+      { search && <GrClose className="text-light-slate-9" fontSize={16} onClick={emptySearch}/> }
```
Comment (brandonroberts) [inline components/atoms/Search/search.tsx:45]: nit: change click handler to `handleEmpty`

```suggestion
      { search && <GrClose className="text-light-slate-9" fontSize={16} onClick={emptySearch}/> }
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]
LGTM

**Review by bdougie — APPROVED** [review]

**Comment by bdougie** [comment]
The deploy previews were blocked because this was a fork. You can open PRs locally in the repo @Deadreyo

**Review by SunGoldTech — APPROVED** [review]
LGTM

**Pull request merged**